From 079682000e035c9ccb74c8364336961b0f7c67d6 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 10 Jul 2018 17:54:42 +0300
Subject: [PATCH] board: habana: attempt graceful remote shutdown

Shutting down the power to the pciehost while it runs an o/s is
a bit harsh...
Attempt a more graceful shutdown, and anyway cut the power off.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/common/remote_power b/board/habanalabs/common/remote_power
index 8b0f25a..97fbca0 100755
--- a/board/habanalabs/common/remote_power
+++ b/board/habanalabs/common/remote_power
@@ -5,9 +5,10 @@
 
 cd `dirname $0`
 
-if [ $# -eq 2 ]; then
+if [ $# -ge 2 ]; then
 	board=$1
 	action=$2
+	pciehost=${3:-monster5}
 else
 	echo "error - usage: $0 <board name> <-l | on | off>"
 	exit 1
@@ -65,6 +66,9 @@
 	$on
 	;;
 off)
+	sysrq=/proc/sysrq-trigger
+	sshpass -p123456 ssh root@$pciehost "echo s > $sysrq && echo u > $sysrq && sleep 3 && echo o > $sysrq"
+	sleep 10
 	$off
 	;;
 *)
